Shared/Samples/Create load report/CreateLoadReportView.Model.swift

```diff
@@ -0,0 +1,299 @@
+                    default:
+                        break
+                    }
+                }
+                summaries[phase] = PhaseSummary(
+                    totalCustomers: totalCustomers,
+                    totalLoad: totalLoad
+                )
+            }
+            resetEnabled = true
+            statusText = nil
+        }
+        
+        func addPhase(_ phase: CodedValue) {
+            includedPhases.append(phase)
+            excludedPhases = excludedPhases.filter { $0 != phase }
+            sortPhases()
+            runEnabled = true
+        }
+        
+        func deletePhase(_ phase: CodedValue) {
+            excludedPhases.append(phase)
+            includedPhases = includedPhases.filter { $0 != phase }
+            sortPhases()
+            summaries[phase] = nil
+        }
+        
+        func deletePhase(atOffsets indexSet: IndexSet) {
+            includedPhases.remove(atOffsets: indexSet)
+            let union = Set(includedPhases).union(Set(excludedPhases))
+            let subtraction = Set(allPhases).subtracting(Array(union))
+            if let phase = subtraction.first {
+                deletePhase(phase)
+            }
+        }
+        
+        func sortPhases() {
+            includedPhases = includedPhases.sorted { $0.name < $1.name }
+            excludedPhases = excludedPhases.sorted { $0.name < $1.name }
+        }
+        
+        func summaryForPhase(_ phase: CodedValue) -> String? {
+            if let summary = summaries[phase] {
+                let formattedString = String(
+                    format: "C: %@  L: %@",
+                    numberFormatter.string(from: summary.totalCustomers as NSNumber)!,
+                    numberFormatter.string(from: summary.totalLoad as NSNumber)!
+                )
+                return formattedString
+            } else {
+                return nil
+            }
+        }
+        
+        func reset() {
+            summaries.removeAll()
+            resetEnabled = false
+        }
+        
+        deinit {
+            ArcGISEnvironment.authenticationManager.arcGISCredentialStore.removeAll()
+        }
+    }
+}
+
+private extension ArcGISCredential {
+    /// The public credentials for the data in this sample.
+    /// - Note: Never hardcode login information in a production application. This is done solely
+    /// for the sake of the sample.
+    static var publicSample: ArcGISCredential {
+        get async throws {
+            try await TokenCredential.credential(
+                for: .utilityNetwork,
+                username: "viewer01",
+                password: "I68VGU^nMurF"
+            )
+        }
+    }
+}
+
+extension CodedValue: Equatable {
+    public static func == (lhs: CodedValue, rhs: CodedValue) -> Bool {
+        lhs.name == rhs.name
+    }
+}
+
+extension CodedValue: Hashable {
+    public func hash(into hasher: inout Hasher) {
+        hasher.combine(name)
+    }
+}
```

Comment:
I'm not sure what is the best practice, but it seems to me that adding public extension to ArcGIS types would have side effects. We removed the map and graphics overlay's `ObservableObject` conformance in 200.1 and I think it is the same case here.
 
ref https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/pull/223

Reply:
Okay is there another way to add the conformance? I added it to create the dictionary.

<img width="1051" alt="Screenshot 2023-07-07 at 2 31 22 PM" src="https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/117859673/fb275e9e-53ac-4122-a273-e8f0cfb912b7">

Reply:
Let's wait for Ryan's thoughts. All I can think of are hacky ways to store the coded value with a hashable dict key. 🤕